File: ui/components/left_pane.go
package components

import (
	"fmt"
	"strings"

	tea "github.com/charmbracelet/bubbletea"
)

// LeftPane represents the left pane with file explorer and knowledge graph
type LeftPane struct {
	width     int
	height    int
	viewMode  string // "files" or "knowledge"
	cursor    int
	items     []string
	knowledge []KnowledgeItem
}

// KnowledgeItem represents a knowledge graph item
type KnowledgeItem struct {
	Entity      string
	Type        string
	Connections []string
}

// NewLeftPane creates a new left pane
func NewLeftPane() *LeftPane {
	return &LeftPane{
		viewMode: "files",
		items: []string{
			"📁 agents/",
			"📁 api/",
			"📁 database/",
			"📁 finance/",
			"📁 knowledge/",
			"📁 memory/",
			"📁 orchestrator/",
			"📁 ui/",
		},
		knowledge: []KnowledgeItem{
			{Entity: "Megamind System", Type: "System", Connections: []string{"Memory", "Finance", "Knowledge"}},
			{Entity: "Memory Store", Type: "Component", Connections: []string{"Feed", "Actions", "API"}},
			{Entity: "Finance Manager", Type: "Component", Connections: []string{"Banking", "Crypto", "Analytics"}},
			{Entity: "Knowledge Graph", Type: "Component", Connections: []string{"Patterns", "Insights", "Queries"}},
			{Entity: "Butler Interface", Type: "UI", Connections: []string{"Three-pane", "Navigation", "Theme"}},
			{Entity: "Contract System", Type: "Orchestration", Connections: []string{"Captain", "Worker", "Tester"}},
		},
	}
}

// Init initializes the left pane
func (lp *LeftPane) Init() tea.Cmd {
	return nil
}

// Update handles updates for the left pane
func (lp *LeftPane) Update(msg tea.Msg) (*LeftPane, tea.Cmd) {
	switch msg := msg.(type) {
	case tea.KeyMsg:
		switch msg.String() {
		case "up":
			if lp.cursor > 0 {
				lp.cursor--
			}
		case "down":
			if lp.viewMode == "files" && lp.cursor < len(lp.items)-1 {
				lp.cursor++
			} else if lp.viewMode == "knowledge" && lp.cursor < len(lp.knowledge)-1 {
				lp.cursor++
			}
		case "tab":
			lp.toggleViewMode()
		}
	}
	return lp, nil
}

// View renders the left pane
func (lp *LeftPane) View() string {
	var content string

	if lp.viewMode == "files" {
		content = lp.renderFileExplorer()
	} else {
		content = lp.renderKnowledgeGraph()
	}

	title := "🗂️ Files & Knowledge"
	subtitle := fmt.Sprintf("Mode: %s", strings.Title(lp.viewMode))

	return LeftPaneStyle.Render(fmt.Sprintf("%s\n%s\n\n%s", title, subtitle, content))
}

// SetSize sets the pane size
func (lp *LeftPane) SetSize(width, height int) {
	lp.width = width
	lp.height = height
}

// toggleViewMode switches between file explorer and knowledge graph
func (lp *LeftPane) toggleViewMode() {
	if lp.viewMode == "files" {
		lp.viewMode = "knowledge"
		lp.cursor = 0
	} else {
		lp.viewMode = "files"
		lp.cursor = 0
	}
}

// renderFileExplorer renders the file explorer view
func (lp *LeftPane) renderFileExplorer() string {
	var lines []string

	for i, item := range lp.items {
		if i == lp.cursor {
			lines = append(lines, CursorStyle.Render("> "+item))
		} else {
			lines = append(lines, "  "+item)
		}
	}

	return strings.Join(lines, "\n")
}

// renderKnowledgeGraph renders the knowledge graph view
func (lp *LeftPane) renderKnowledgeGraph() string {
	var lines []string

	for i, item := range lp.knowledge {
		line := fmt.Sprintf("%s (%s)", item.Entity, item.Type)
		if i == lp.cursor {
			lines = append(lines, CursorStyle.Render("> "+line))
		} else {
			lines = append(lines, "  "+line)
		}

		// Add connections
		connections := strings.Join(item.Connections, ", ")
		if i == lp.cursor {
			lines = append(lines, "    "+ConnectionStyle.Render(connections))
		} else {
			lines = append(lines, "    "+connections)
		}
		lines = append(lines, "")
	}

	return strings.Join(lines, "\n")
}

File: ui/components/left_pane.py
"""
Left Pane - File Explorer + Knowledge Graph

Displays file system navigation and knowledge graph relationships.
"""

from typing import List, Dict, Any, Optional
from pathlib import Path
from rich.panel import Panel
from rich.tree import Tree
from rich.text import Text
from rich.table import Table
from rich.columns import Columns


class LeftPane:
    """Left pane showing file explorer and knowledge graph."""
    
    def __init__(self, memory_connector):
        self.memory_connector = memory_connector
        self.current_path = Path.cwd()
        self.selected_item = None
        self.view_mode = "files"  # "files" or "knowledge"
        
    def render(self) -> Panel:
        """Render the left pane content."""
        if self.view_mode == "files":
            content = self._render_file_explorer()
        else:
            content = self._render_knowledge_graph()
            
        return Panel(
            content,
            title="🗂️ Files & Knowledge",
            border_style="blue",
            subtitle=f"Mode: {self.view_mode.title()}"
        )
    
    def _render_file_explorer(self) -> Tree:
        """Render the file explorer tree."""
        tree = Tree("📁 Project Root", style="bold blue")
        
        # Add main directories
        project_root = Path.cwd()
        main_dirs = [
            "agents", "api", "database", "finance", 
            "knowledge", "memory", "orchestrator", "ui"
        ]
        
        for dir_name in main_dirs:
            dir_path = project_root / dir_name
            if dir_path.exists():
                dir_node = tree.add(f"📁 {dir_name}", style="cyan")
                
                # Add some key files in each directory
                if dir_name == "agents":
                    dir_node.add("📄 contracts.json", style="green")
                    dir_node.add("📄 agents.md", style="green")
                elif dir_name == "ui":
                    dir_node.add("🐍 main.py", style="yellow")
                    dir_node.add("📄 requirements.txt", style="yellow")
                elif dir_name == "memory":
                    dir_node.add("🗄️ memory_store.py", style="green")
                    dir_node.add("📊 megamind_memory.db", style="green")
                elif dir_name == "finance":
                    dir_node.add("💰 finance_manager.py", style="green")
                    dir_node.add("📈 finance_data.db", style="green")
        
        return tree
    
    def _render_knowledge_graph(self) -> Table:
        """Render the knowledge graph relationships."""
        table = Table(title="🧠 Knowledge Graph", show_header=True)
        table.add_column("Entity", style="cyan", no_wrap=True)
        table.add_column("Type", style="magenta")
        table.add_column("Connections", style="green")
        
        # Sample knowledge graph data
        knowledge_data = [
            ("Megamind System", "System", "Memory, Finance, Knowledge"),
            ("Memory Store", "Component", "Feed, Actions, API"),
            ("Finance Manager", "Component", "Banking, Crypto, Analytics"),
            ("Knowledge Graph", "Component", "Patterns, Insights, Queries"),
            ("Butler Interface", "UI", "Three-pane, Navigation, Theme"),
            ("Contract System", "Orchestration", "Captain, Worker, Tester"),
        ]
        
        for entity, entity_type, connections in knowledge_data:
            table.add_row(entity, entity_type, connections)
            
        return table
    
    def toggle_view_mode(self):
        """Toggle between file explorer and knowledge graph views."""
        self.view_mode = "knowledge" if self.view_mode == "files" else "files"
    
    def navigate_to(self, path: str):
        """Navigate to a specific path in file explorer."""
        new_path = Path(path)
        if new_path.exists():
            self.current_path = new_path
    
    def select_item(self, item: str):
        """Select an item in the current view."""
        self.selected_item = item
